### DISCUSSION TITLE
airflow db init mysql error

### DISCUSSION THREAD

**Discussion by NQ31** [apache/airflow#34334] [General]
### Apache Airflow version

2.7.1

### What happened

sqlalchemy.exc.OperationalError: (MySQLdb._exceptions.OperationalError) (1067, "Invalid default value for 'updated_at'")
[SQL: 
CREATE TABLE dataset (
        id INTEGER NOT NULL AUTO_INCREMENT, 
        uri VARCHAR(3000) COLLATE latin1_general_cs NOT NULL, 
        extra JSON NOT NULL, 
        created_at TIMESTAMP(6) NOT NULL, 
        updated_at TIMESTAMP(6) NOT NULL, 
        is_orphaned BOOL NOT NULL DEFAULT '0', 
        CONSTRAINT dataset_pkey PRIMARY KEY (id)
)

]
(Background on this error at: https://sqlalche.me/e/14/e3q8)

### What you think should happen instead

我认为应该是初始化的时候，有bug

### How to reproduce

重新初始化

### Operating System

centos7.9

### Deployment

Official Apache Airflow Helm Chart

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by Taragolis** [comment]
Which MySQL version do you use? 

Did you follow all advices listed in [Setting up a MySQL Database](https://airflow.apache.org/docs/apache-airflow/stable/howto/set-up-database.html#setting-up-a-mysql-database), there is note exists about similar error (it could be other cases also)

> [!NOTE]  
> In strict mode, MySQL doesn’t allow 0000-00-00 as a valid date. Then you might get errors like "Invalid default value for 'end_date'" in some cases (some Airflow tables use 0000-00-00 00:00:00 as timestamp field default value). To avoid this error, you could disable NO_ZERO_DATE mode on you MySQL server. Read https://stackoverflow.com/questions/9192027/invalid-default-value-for-create-date-timestamp-field for how to disable it. See [SQL Mode - NO_ZERO_DATE](https://dev.mysql.com/doc/refman/5.7/en/sql-mode.html#sqlmode_no_zero_date) for more information.
